script.js

```diff
@@ -114,4 +99,31 @@ function calculatePath(startPoint) {
   }
   fieldStrength = [fieldStrength[0] / Math.sqrt(fieldStrength[0] ** 2 + fieldStrength[1] ** 2), fieldStrength[1] / Math.sqrt(fieldStrength[0] ** 2 + fieldStrength[1] ** 2)];
   return [startPoint].concat(calculatePath([startPoint[0] + fieldStrength[0], startPoint[1] + fieldStrength[1], startPoint[2]]));
+}
+
+function drawPath(paths, i) {
+  ctx = document.getElementById('lines').getContext('2d');
+  ctx.beginPath();
+  if (Math.sign(paths[i][0][2]) === 1) {
+    ctx.strokeStyle = "#900000"
+  } else {
+    ctx.strokeStyle = "#000090"
+  }
+  ctx.moveTo(paths[i][0][0], paths[i][0][1]);
+  for (var j = 0; j < paths[i].length; j++) {
+    ctx.lineTo(paths[i][j][0], paths[i][j][1]);
+  }
+  ctx.stroke();
+}
+
+window.onresize = function () {
+  updateSliders();
+  drawFieldLines();
+}
+
+function updateSliders() {
+  for (let i = 0; i < sliders.length; i += 2) {
+    sliders[i + 1][0].max = window.innerWidth;
```

Comment:
Two comments about this:

1. This seems to be the only place you use the odd elements of the `sliders` array. It might be easier to just keep the `div`s in the array and use the DOM to get at the sliders which will be the `childNode`s of the `div`.
2. I notice that here you set the `max` to be the full value of `innerWidth` and `innerHeight` but when you made the sliders you set the max to `innerWidth - 10` and `innerHeight - 10`. Should probably be consistent, no? One way to achieve that would be to use this function to set the maxes as soon as you make the sliders.